leader-election-impl/src/test/java/com/palantir/leader/proxy/AwaitingLeadershipProxyTest.java

```
@@ -183,6 +192,72 @@ public void shouldAllowObjectMethodsWhenNotLeading() {
         assertThat(proxy.toString()).startsWith("com.palantir.leader.proxy.AwaitingLeadershipProxy@");
     }
 
+    @Test
+    public void shouldThrowIfNotLeading() {
+        Callable<Void> delegate = () -> {
```

Thanks for adding the tests! We're still not testing the case where we throw rather than returning a failed future (i.e., the mistake in the previous iteration), because the handleInvocation method unwraps the future anyway if the method under the proxy does not return a ListenableFuture. As a result, in the current test framework, the cases of throwing vs returning a failed future are not visibly different.

That said, from a review perspective, I'm happy with the change. If you want to go ahead with trying to get our tests to repro the above failure mode, please feel free and I can give it another review.